lib/openhab/core/dsl/items/datetime_item.rb

```diff
@@ -0,0 +1,97 @@
+# frozen_string_literal: true
+
+require 'forwardable'
+require 'java'
+require 'time'
+require 'openhab/core/dsl/types/datetime'
+
+module OpenHAB
+  module Core
+    module DSL
+      module Items
+        #
+        # Delegation to OpenHAB DateTime Item
+        #
+        class DateTimeItem
+          extend Forwardable
+          include Comparable
+
+          def_delegator :@datetime_item, :to_s
+
+          #
+          # Create a new DateTimeItem
+          #
+          # @param [Java::org::openhab::core::libarary::items::DateTimeItem] datetime_item Openhab DateTimeItem to
+          # delegate to
+          #
+          def initialize(datetime_item)
+            @datetime_item = datetime_item
+          end
+
+          #
+          # Return an instance of DateTime that wraps the DateTimeItem's state
+          #
+          # @return [OpenHAB::Core::DSL::Types::DateTime] Wrapper for the Item's state, or nil if it has no state
+          #
+          def to_dt
+            OpenHAB::Core::DSL::Types::DateTime.new(@datetime_item.state) if state?
+          end
+
+          #
+          # Compare the Item's state to another Item or object that can be compared
+          #
+          # @param [Object] other Other objet to compare against
+          #
+          # @return [Integer] -1, 0 or 1 depending on the result of the comparison
+          #
+          def <=>(other)
+            return unless state?
```

Comment:
Just a question (I'm still learning). Where does this `state?` come from?

Reply:
`lib/openhab/core/dsl/monkey_patch/items/items.rb` monkeypatches the openhab Item classes with some additional methods that can be used. The reason it works here is because it gets delegated to the DateTimeItem via `method_missing`